### PR TITLE
layers: Improve checking for external Fence and Semaphore VUIDs

### LINKED ISSUE #4402: Potential race for sync fd fence export and retire
**Describe the Issue**
sync_fd external fence export implies a fence reset operation, which has been reflected by `PostCallRecordGetFenceFdKHR`. However, it misses to remove this exported sync_fd fence from the associated submission. If there's another fence been waited, the prior submissions will be processed and the associated fence will be set to retired. Then the below VU violation shows up upon the next submission using the sync_fd fence.

**Valid Usage ID**
Validation Error: [ VUID-vkQueueSubmit-fence-00063 ] Object 0: handle = 0xe7e6d0000000000f, type = VK_OBJECT_TYPE_UNKNOWN; | MessageID = 0x6f35a2b3 | vkQueueSubmit(): VkNonDispatchableHandle 0xe7e6d0000000000f[] submitted in SIGNALED state.  Fences must be reset before being submitted The Vulkan spec states: If fence is not VK_NULL_HANDLE, fence must be unsignaled (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkQueueSubmit-fence-00063)

**Environment:**
 - OS: linux

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
layers: Fix FENCE_STATE::Reset()

Handle VK_FENCE_IMPORT_TEMPORARY_BIT imported fences correctly.

Fixes #4402

 stateless: Add more external Semaphore and Fence VUIDs

    VUID-VkSemaphoreGetFdInfoKHR-handleType-01136
    VUID-VkImportSemaphoreFdInfoKHR-handleType-01143
    VUID-VkImportSemaphoreFdInfoKHR-handleType-07307
    VUID-VkFenceGetFdInfoKHR-handleType-01456
    VUID-VkImportFenceFdInfoKHR-handleType-01464
    VUID-VkImportFenceFdInfoKHR-handleType-07306
    VUID-VkImportSemaphoreWin32HandleInfoKHR-handleType-01140
    VUID-VkImportSemaphoreWin32HandleInfoKHR-handleType-01466
    VUID-VkImportSemaphoreWin32HandleInfoKHR-handleType-01469
    VUID-VkSemaphoreGetWin32HandleInfoKHR-handleType-01131
    VUID-VkImportFenceWin32HandleInfoKHR-handleType-01457
    VUID-VkImportFenceWin32HandleInfoKHR-handleType-01459
    VUID-VkImportFenceWin32HandleInfoKHR-handle-01462
    VUID-VkFenceGetWin32HandleInfoKHR-handleType-01452

core: Add more external Semaphore and Fence VUIDs

    VUID-VkImportSemaphoreWin32HandleInfoKHR-flags-03322
    VUID-VkSemaphoreGetWin32HandleInfoKHR-handleType-01126
    VUID-VkImportSemaphoreFdInfoKHR-flags-03323
    VUID-VkSemaphoreGetFdInfoKHR-handleType-01132
    VUID-VkSemaphoreGetFdInfoKHR-handleType-03253
    VUID-VkSemaphoreGetFdInfoKHR-handleType-03254
    VUID-vkImportFenceWin32HandleKHR-fence-04448
    VUID-VkFenceGetWin32HandleInfoKHR-handleType-01448
    VUID-VkFenceGetFdInfoKHR-handleType-01453
    VUID-VkFenceGetFdInfoKHR-handleType-01454

tests: Improve coverage of external Semaphores and Fences

Make it easier to write cross platform test code and add coverage for the recently added VUIDs.

### CODE REVIEW

**Review by Tony-LunarG — CHANGES_REQUESTED** [review]
This is great, both from a validation and tests POV.  Do the external import/export tests run anywhere?  Everywhere I looked they were being skipped.  One VUID is misnamed.

**Comment by jeremyg-lunarg** [comment]
> This is great, both from a validation and tests POV. Do the external import/export tests run anywhere? Everywhere I looked they were being skipped. One VUID is misnamed.

Yes they run for me now on both windows and linux.

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -7126,22 +7126,147 @@ bool StatelessValidation::manual_PreCallValidateCreateSamplerYcbcrConversionKHR(
...
+
+    static constexpr auto kNameAllowedTypes =
+        VK_EXTERNAL_SEMAPHORE_HANDLE_TYPE_OPAQUE_WIN32_BIT | VK_EXTERNAL_SEMAPHORE_HANDLE_TYPE_D3D12_FENCE_BIT;
+    if ((info->handleType & kNameAllowedTypes) == 0 && info->name) {
+        skip |= LogError(info->semaphore, "VUID-VkImportSemaphoreWin32HandleInfoKHR-handleType-01466",
+                         "%s(): name (%p) must be NULL if handleType is %s", func_name, info->name,
+                         string_VkExternalSemaphoreHandleTypeFlagBits(info->handleType));
+    }
+    if (info->handle && info->name) {
+        skip |= LogError(info->semaphore, "VUID-VkImportSemaphoreWin32HandleInfoKHR-handleType-01469",
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply layers/parameter_validation_utils.cpp]: doh. Fixed now. I just ran scripts/vk_validation_stats.py  build/Vulkan-Headers/registry/validusage.json -c and I don't see any more wrong ones.

**Comment by jeremyg-lunarg** [comment]
> > This is great, both from a validation and tests POV. Do the external import/export tests run anywhere? Everywhere I looked they were being skipped. One VUID is misnamed.
> 
> Yes they run for me now on both windows and linux.

Also, the new SyncFd ones run on android.

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**